### ISSUE TITLE
http: panic serving 10.254.237.226:42805: sync: Unlock of unlocked RWMutex

### ISSUE THREAD

**Issue by ror6ax** [hashicorp/vault#2346]
Currently figuring out the reason of this, but certainly has to do with HA cluster_address or redirect_addr
introduced.

Vault v0.6.4

```
2017/02/07 13:23:36.343576 [INFO ] core: security barrier initialized: shares=1 threshold=1
2017/02/07 13:23:36.346268 [ERROR] core: failed to get cluster details: error=decryption failed: cipher: message authentication failed
2017/02/07 13:23:36.346289 [ERROR] core: cluster setup failed during init: error=decryption failed: cipher: message authentication failed
2017-02-07 13:23:36.346381 I | http: panic serving 10.254.237.226:42805: sync: Unlock of unlocked RWMutex
goroutine 39 [running]:
net/http.(*conn).serve.func1(0xc4201ca200)
	/usr/local/go/src/net/http/server.go:1491 +0x12a
panic(0x151bf00, 0xc420203720)
	/usr/local/go/src/runtime/panic.go:458 +0x243
sync.(*RWMutex).Unlock(0xc420058e80)
	/usr/local/go/src/sync/rwmutex.go:118 +0xef
github.com/hashicorp/vault/vault.(*Core).Initialize(0xc420058e00, 0xc42061bb00, 0x0, 0x23577c0, 0xc420203670)
	/ephemeral/jenkins-agent/workspace/src/github.com/hashicorp/vault/vault/init.go:192 +0xc75
github.com/hashicorp/vault/http.handleSysInitPut(0xc420058e00, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/ephemeral/jenkins-agent/workspace/src/github.com/hashicorp/vault/http/sys_init.go:102 +0x3b5
github.com/hashicorp/vault/http.handleSysInit.func1(0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/ephemeral/jenkins-agent/workspace/src/github.com/hashicorp/vault/http/sys_init.go:19 +0xeb
net/http.HandlerFunc.ServeHTTP(0xc4200fb6f0, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xc420207e90, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/usr/local/go/src/net/http/server.go:2022 +0x7f
github.com/hashicorp/vault/http.wrapHelpHandler.func1(0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/ephemeral/jenkins-agent/workspace/src/github.com/hashicorp/vault/http/help.go:18 +0xd8
net/http.HandlerFunc.ServeHTTP(0xc4201c94a0, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
github.com/hashicorp/vault/http.wrapGenericHandler.func1(0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/ephemeral/jenkins-agent/workspace/src/github.com/hashicorp/vault/http/handler.go:85 +0xb1
net/http.HandlerFunc.ServeHTTP(0xc4201c94c0, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
net/http.serverHandler.ServeHTTP(0xc4201caa80, 0x2366840, 0xc420126f70, 0xc4200ec2d0)
	/usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc4201ca200, 0x2367ac0, 0xc420253200)
	/usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2293 +0x44d

```

**Comment by ror6ax** [comment]
My config file is as follows: {"backend": {"zookeeper": {"address":"master.mesos:2181", "redirect_addr":"http://0.0.0.0:6379" }}, "listener": {"tcp": {"address":"0.0.0.0:8200","cluster_address":"0.0.0.0:3900","tls_disable":1}}}

**Comment by vishalnayak** [comment]
Addressed in mailing list (https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/vault-tool/vShISC1KBnY/4FbEDv-rDgAJ). Fixed by 943cfd9b2407eab2a487c16e8c2e5bc244a6d597.

**Comment by jefferai** [comment]
Whoops, thanks @vishalnayak  -- I didn't notice this when I was looking at the mailing list.
